src/components/UI/Alert/ImageUploaderDialog.vue

```diff
@@ -0,0 +1,180 @@
+      imageToPreview: null, // the URL of the image that needs to be previewed
+      localImageData: null, // the data of the image that has been uploaded, saved locally
+      // styling class for the image preview div
+      imagePreviewClass:
+        "object-contain h-full w-full border border-dashed border-gray-700",
+    };
+  },
+
+  created() {
+    if (this.existingImage != null) {
+      // if there is an existing image, it needs to be previewed when the dialog opens up
+      this.showImagePreview = true;
+      this.imageToPreview = this.existingImage;
+    }
+  },
+
+  computed: {
+    uploaderInputClass() {
+      // styling class for the input element that handles uploading
+      return [
+        { hidden: this.showImagePreview },
+        "w-full h-full opacity-0 absolute inset-0 cursor-pointer",
+      ];
+    },
+    deleteButtonTitleConfig() {
+      // title config for the delete button
+      return {
+        value: "Delete",
+        class: "text-black hover:text-white text-md font-bold",
+      };
+    },
+    deleteButtonClass() {
+      // class for the delete button
+      return "bg-gray-300 hover:bg-red-500 p-1 px-2 rounded-md shadow-xl disabled:opacity-50 disabled:pointer-events-none h-10 place-self-center";
+    },
+    doneButtonTitleConfig() {
+      // title config for the done button
+      return {
+        value: "Done",
+        class: "text-white text-md font-bold",
+      };
+    },
+    doneButtonClass() {
+      // class for the done button
+      return "bg-primary hover:bg-primary-hover p-1 px-2 rounded-md shadow-xl disabled:opacity-50 disabled:pointer-events-none h-10 place-self-center";
+    },
+  },
+
+  methods: {
+    ...Utilities,
+    loadAndPreviewImage(imageInfo) {
+      // save the image info locally
+      // extract the base64 URL from the info and
+      // use it to show the preview
+      if (imageInfo != undefined) {
+        this.localImageData = imageInfo;
+        this.imageToPreview = this.localImageData.dataUrl;
+        this.showImagePreview = true;
+      }
+    },
+    closeDialog() {
+      // emit the event that the dialog has to be closed
+      this.$emit("close-dialog");
+    },
+    deleteAndUnsetImage() {
+      // invoked when the user clicks the delete button
+      // hide the image preview
+      this.showImagePreview = false;
+      this.imageToPreview = null;
+
+      // if the image was saved locally only, purge the local data
+      // else if the image was saved in the DB, emit an event
+      // (image deletion and unlinking is handled separately)
+      if (this.existingImage == null) this.localImageData = null;
+      else this.$emit("delete-and-unlink-image");
+    },
+    submitImage() {
+      // when a locally uploaded image needs to be submitted to the DB,
+      // emit an event with the image file as a payload
+      if (this.localImageData != null)
+        this.$emit("image-selected", this.localImageData.file);
```

Comment:
if someone is not updating an image - say image was already there, they selected on edit, didn't do anything, clicked done - will this be invoked? - no right?